### DISCUSSION TITLE
Contribution Reward Program

### DISCUSSION THREAD

**Discussion by Balianos** [capsule-corp-ternoa/ternoa-hub#65] [Ideas]
TODO

1) Public formation session on coding
2) Express projects/repos that needs to be challenged and contributed
3) Identify Qualitative Metrics (commits, code lines, Pull Requests validations and reviews…)
4) Apply a Metrics rating scale for rewards
5) Rewarding in $CAPS

Dig this Suggested Platforms/Projects to use for Contribution Reward/Bounty Program : 
- LinearB (https://linearb.io/)
- Gitcoin (https://gitcoin.co/)
- Starfish (https://github.com/indeedeng/starfish)

Find a portal name : **Ternoa Developer Portal** (to validate)

-----------------------------------------------------------------------------------------------------------------------------------

Ternoa and Capsule Corp Labs will announce soon its first Contribution Rewards program, designed to support innovation that contributes to the growth and improvement of the Ternoa ecosystem on GitHub. 

This first Reward program will run for 6 months (or when the pool of $CAPS awards engaged is exhausted). 
Contributors' submissions will be listed on the Ternoa Developer Portal to showcase your work. We would also love to get feedback from the community on ideas that could inspire future iterations of our developer program. The contest is open to developers, enthusiasts, and students !

### How to contribute to Ternoa

As part of its vision to cultivate a growing and active qualitative blockchain development community, Ternoa is committed to support this community in its quest for the innovation opportunities that exist within the Ternoa Blockchain ecosystem. 

Ternoa is interested in NFT use cases through the new Capsule concept invented by Capsule Corp. But above all, Ternoa is open to any other field of innovation around blockchain. Applications will be judged on creativity, usefulness, quality and accuracy of the code.

**Do you like organizing ?**

- Link to duplicate issues, and suggest new issue labels, to keep things organized
- Go through open issues and suggest closing old ones
- Ask clarifying questions on recently opened issues to move the discussion forward

**Do you like to code?**

- Find an open issue to contribute
- Ask if you can help write a new feature
- Automate project setup or dependencies (update and check new versions of plugins dependencies in the Ternoa Stack / SDK)
- Improve tooling and testing

**Do you like helping people ?**

- Answer questions about the project on our Github or e.g., Stack Overflow, Reddit or Polkahub
- Answer questions for people on open issues
- Help moderate the discussion boards or conversation channels

**Do you like helping others code ?**

- Review code on other people’s submissions
- Write tutorials about how Ternoa SDK can be used
- Offer to mentor another contributor

You don’t just have to work on software projects !

While Ternoa often refers to blockchain or coding, you can collaborate on just about anything. There are books, recipes, lists, and classes that get developed as open source projects.

Even if you’re a software developer, working on the Ternoa documentation can help you get started in open source. It’s often less intimidating to work on projects that don’t involve code, and the process of collaboration will build your confidence and experience.

------------------------------------------

### Award Program to highlight best Ternoa contributor profiles

- Top Ambassador : A champion for the Ternoa space, this person helps spread awareness of Capsule Corp and its projects. The Top Ambassador leverages multiple platforms, both online as well as speaking engagements, driving interest and excitement around the ecosystem.
- Top Committer : This will recognize excellence in technical contributions to Capsule Corp and its projects. The Ternoa Top Committer has made key commits to projects and, more importantly, contributes in a way that benefits the project neutrally as a whole.
- Top End User : The special award for the Ternoa Developer Portal, representatives in recognition of major contributions to the Ternoa ecosystem.
- Top Documentarian : This award recognizes excellence in documentation contributions to Ternoa Documentation and Capsule Corp projects. Excellent technical documentation is one of the best ways projects can lower the barrier to contribution.
- Thugz Award : A special bonus award for the community contributors

**Comment by Balianos** [comment]
Questions to ask ourselves if Ternoa has the perfect Github repo : 

Do we have a Complete Community Profile ?
What is the Code of Conduct for the Ternoa Dev Portal ?
What is the Ternoa Contributing Guide and How to create it ?
Do we have to create Custom Issue Templates in GitHub ?
Do we have to create a Pull Request Template ?
Do we have to Socialize Ternoa GitHub Repository ?
Do we have to enable the Sponsor Button ?
Do we have to make Ternoa repository a template repository for other projects ?

**Comment by jeemclr** [comment]
@Balianos Some github bot could be interesting : 
- https://allcontributors.org/#
- https://github.com/marketplace/actions/reward-contributors
- https://github.com/marketplace/actions/bonusly-for-github

**Comment by Balianos** [comment]
First Needs : 
- Having an automated list of contributors
- Explain what the idea of the Ternoa Doc
- Start talking about contributions
